File: 1603. Design Parking System.cpp
class ParkingSystem {
public:
int bigSlots = 0, medSlots = 0, smallSlots = 0;
    ParkingSystem(int big, int medium, int small) {
        // no.of slots available
        bigSlots = big;
        medSlots = medium;
        smallSlots = small;
    }
    
    bool addCar(int carType) {
        // store the car and decrease no.of stores avaialble by one for each car
        if (carType == 1 && bigSlots)
            {
                bigSlots--;
                return true;
            }
        else if(carType==2 && medSlots)
        {
            medSlots--;
            return true;
        }else if(carType==3 && smallSlots)
        {
            smallSlots--;
            return true;
        }
        // if no available stores
        return false;
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */

File: 1603. Design Parking System.py
class ParkingSystem:
    def __init__(self, big: int, medium: int, small: int):
        # no.of slots available
        self.bigSlots, self.medSlots, self.smallSlots = big, medium, small

    def addCar(self, carType: int) -> bool:
        # store the car and decrease no.of stores avaialble by one for each car
        if carType==1 and self.bigSlots:
            self.bigSlots-=1
            return True
        elif carType==2 and self.medSlots:
            self.medSlots-=1
            return True
        elif carType==3 and self.smallSlots:
            self.smallSlots-=1
            return True

        # if no available stores
        return False

        


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)